### ISSUE TITLE
HTTP Error 404: Not Found. requested URL: 'http://sncosmo.github.io/data/models/pierel/salt2.tar.gz'

### ISSUE THREAD

**Issue by tallamjr** [sncosmo/sncosmo#240]
Recently we have been getting this error when running CI checks on [`snmachine`](https://github.com/LSSTDESC/snmachine):
```
test/snfeatures_test.py:79: in fit_templates
    nprocesses=nprocesses, seed=42)
../../../miniconda/envs/snmachine/lib/python3.6/site-packages/snmachine/snfeatures.py:723: in extract_features
    self.model=sncosmo.Model(self.templates[mod_name])
../../../miniconda/envs/snmachine/lib/python3.6/site-packages/sncosmo/models.py:1053: in __init__
    self._source = get_source(source, copy=True)
../../../miniconda/envs/snmachine/lib/python3.6/site-packages/sncosmo/models.py:89: in get_source
    return cp(_SOURCES.retrieve(name, version=version))
../../../miniconda/envs/snmachine/lib/python3.6/site-packages/sncosmo/_registry.py:179: in retrieve
    version=latest_version)
../../../miniconda/envs/snmachine/lib/python3.6/site-packages/sncosmo/builtins.py:448: in load_salt2model
    abspath = DATADIR.abspath(relpath, isdir=True)
../../../miniconda/envs/snmachine/lib/python3.6/site-packages/sncosmo/utils.py:466: in abspath
    download_dir(url, os.path.dirname(abspath))
../../../miniconda/envs/snmachine/lib/python3.6/site-packages/sncosmo/utils.py:359: in download_dir
    _download_file(remote_url, buf)
../../../miniconda/envs/snmachine/lib/python3.6/site-packages/sncosmo/utils.py:236: in _download_file
    with closing(urlopen(req, timeout=timeout)) as remote:
../../../miniconda/envs/snmachine/lib/python3.6/urllib/request.py:223: in urlopen
    return opener.open(url, data, timeout)
../../../miniconda/envs/snmachine/lib/python3.6/urllib/request.py:532: in open
    response = meth(req, response)
../../../miniconda/envs/snmachine/lib/python3.6/urllib/request.py:642: in http_response
    'http', request, response, code, msg, hdrs)
../../../miniconda/envs/snmachine/lib/python3.6/urllib/request.py:570: in error
    return self._call_chain(*args)
../../../miniconda/envs/snmachine/lib/python3.6/urllib/request.py:504: in _call_chain
    result = func(*args)
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

self = <urllib.request.HTTPDefaultErrorHandler object at 0x7fb549943898>
req = <urllib.request.Request object at 0x7fb5499ac400>
fp = <http.client.HTTPResponse object at 0x7fb54993f898>, code = 404
msg = 'Not Found', hdrs = <http.client.HTTPMessage object at 0x7fb5499400f0>

    def http_error_default(self, req, fp, code, msg, hdrs):
>       raise HTTPError(req.full_url, code, msg, hdrs, fp)
E       urllib.error.HTTPError: HTTP Error 404: Not Found. requested URL: 'http://sncosmo.github.io/data/models/pierel/salt2.tar.gz'
```
If one tries to look for: http://sncosmo.github.io/data/models/pierel/salt2.tar.gz , it is indeed a broken link. However, if we look for: http://sncosmo.github.io/data/models/pierel/salt2-extended.tar.gz we are able to start the download.

This error occurs with version `v1.8.0` If one were to bump down to the previous release `v1.7.1` this seems to be fine (confirmed with this [check](https://github.com/LSSTDESC/snmachine/pull/157/commits/33cffea91614126d2574d4080e29e61e48488925))

After comparing commits between the two tags, I am yet to figure out what change might be required (perhaps even on our end) in order to pull the correct models.

*Note* In the above error message there is: `self.model=sncosmo.Model(self.templates[mod_name])` where `mod_name` = `'Ia'` which is the key in this dictionary: 
```
self.templates={'Ia':'salt2-extended','salt2':'salt2-extended', 'mlcs2k2':'mlcs2k2', 'II':'nugent-sn2n','IIn':'nugent-sn2n','IIp':'nugent-sn2p', 'IIl':'nugent-sn2l', 'Ibc':'nugent-sn1bc',  'Ib':'nugent-sn1bc',  'Ic':'nugent-sn1bc'}
```

**Comment by kbarbary** [comment]
Thanks for reporting! Discussion in #241. Longer term we need a test that downloads all remote data files, to make sure they all work.

**Comment by kbarbary** [comment]
Fixed in v1.8.2 and v2.0.0, which have now been released. There is also now a script that will download all remote data to a test directory, `check-all-builtins`.
